testsuite/MDAnalysisTests/topology/test_pdb.py

```diff
@@ -244,3 +247,102 @@ def test_PDB_metals():
+HETATM 1613  S   DMS A 102      22.157  39.211  12.217  1.00 27.26           S
+HETATM 1614  O   DMS A 102      20.622  38.811  11.702  1.00 25.51           O
+HETATM 1615  C1  DMS A 102      22.715  40.764  11.522  1.00 26.00           C
+HETATM 1616  C2  DMS A 102      22.343  39.515  13.971  1.00 25.46           C
+TER    1617
+"""
+
+def test_PDB_elements():
+    """
+    The elements attribute is equipped to obtain the element symbols,
+    if the PDB file contains the element symbol
+    """
+    u = mda.Universe(StringIO(PDB_elements), format='PDB')
+    element_list = np.array(['N', 'C', 'C', 'O', 'C', 'C', 'O', 'N', 'H', 'H', 'H', 'H', 'H', 'H', 'H', 'H', 'CU', 'FE', 'MG', 'S', 'O', 'C', 'C', 'S', 'O', 'C', 'C'], dtype=object)
+    assert np.all(u.atoms.elements == element_list)
+
+
+PDB_missing_ele = """\
+ATOM      1  N   ASN A   1      -8.901   4.127  -0.555  1.00  0.00
+ATOM      2  CA  ASN A   1      -8.608   3.135  -1.618  1.00  0.00
+ATOM      3  C   ASN A   1      -7.117   2.964  -1.897  1.00  0.00
+ATOM      4  O   ASN A   1      -6.634   1.849  -1.758  1.00  0.00
+TER       5
+HETATM    6 CU    CU A   2      03.000  00.000  00.000  1.00 00.00
+HETATM    7 FE    FE A   3      00.000  03.000  00.000  1.00 00.00
+HETATM    8 Mg    Mg A   4      03.000  03.000  03.000  1.00 00.00
+TER       9
+"""
+
+PDB_wrong_ele = """\
+ATOM      1  N   ASN A   1      -8.901   4.127  -0.555  1.00  0.00           N
+ATOM      2  CA  ASN A   1      -8.608   3.135  -1.618  1.00  0.00           C
+ATOM      3  C   ASN A   1      -7.117   2.964  -1.897  1.00  0.00           C
+ATOM      4  O   ASN A   1      -6.634   1.849  -1.758  1.00  0.00           O
+ATOM      5  X   ASN A   1      -9.437   3.396  -2.889  1.00  0.00
+TER       5
+HETATM    6 CU    CU A   2      03.000  00.000  00.000  1.00 00.00          CU
+HETATM    7 FE    FE A   3      00.000  03.000  00.000  1.00 00.00
+HETATM    8 Mg    Mg A   4      03.000  03.000  03.000  1.00 00.00          MG
+TER       9
+"""
+
+def test_missing_elements_warnings():
+    """
+    The test checks whether it asserts the appropriate warning on
+    encountering missing elements column.
+    """
+
+    with pytest.warns(UserWarning) as record:
+        u = mda.Universe(StringIO(PDB_missing_ele), format='PDB')
+
+    assert len(record) == 1
+    assert record[0].message.args[0] == "Element information missing. Guessing elements from atom names : ['N', 'CA', 'C', 'O', 'CU', 'FE', 'Mg']"
+
+
+def test_wrong_elements_warnings():
+    """
+    The test checks whether it fills in the wrong elements appropriately
+    with the atom names where element symbols are missing.
+    """
+    with pytest.warns(UserWarning) as record:
+        u = mda.Universe(StringIO(PDB_wrong_ele), format='PDB')
+    
+    assert len(record) == 2
+    assert record[1].message.args[0] == "Element record found to be non-physical. Guessing element from atom name: ['X', 'FE']"
```

Comment:
As above, please follow PEP8. What's the first warning that you're expecting here?

> "Element record found to be non-physical.

I don't see any non-physical elements in `PDB_wrong_ele`, just missing ones. You may need to reconsider how you guess the elements.

Reply:
ATOM 5 was named X, that was a non-physical element which is returned as a string.